lineage.go

```diff
@@ -5,173 +5,100 @@ import (
 	"sort"
 
 	"cuelang.org/go/cue"
-	terrors "github.com/grafana/thema/errors"
-	"github.com/grafana/thema/internal/util"
+	cerrors "cuelang.org/go/cue/errors"
 )
 
 var (
-	_ Lineage                     = &UnaryLineage{}
+	_ Lineage                     = &baseLineage{}
 	_ ConvergentLineage[Assignee] = &unaryConvLineage[Assignee]{}
 )
 
-// A UnaryLineage is a Go facade over a valid CUE lineage that does not compose
+// A baseLineage is a Go facade over a valid CUE lineage that does not compose
 // other lineage.
-type UnaryLineage struct {
```

Comment:
It was used in [Kindsys](https://github.com/grafana/kindsys/blob/812b9884a375b6cdf20aa9c3f9676dd2b770df8b/somedef.go#L28) library. We need to update that too.